### PR TITLE
add error handler for wasm memory detach

### PR DESCRIPTION
**Pull request by ksyeo1010**
not sure if this is okay, but we can tweak the error message or create a error class for it.

### CODE REVIEW

File: binding/web/src/cheetah.ts [hunk 1]
```diff
@@ -89,15 +89,17 @@ export class Cheetah {
 
   private static _cheetahMutex = new Mutex();
 
+  private _isDetached: boolean = false;
+
   private readonly _transcriptCallback: (cheetahTranscript: CheetahTranscript) => void;
-  private readonly _processErrorCallback?: (error: string) => void;
+  private readonly _processErrorCallback?: (error: Error) => void;
```
Comment (laves) [inline binding/web/src/cheetah.ts:95]: even though this is a good change, I don't think we should change the API until we do our next Cheetah release (which shouldn't be too far away). For now, we may have to live with a string

File: binding/web/src/cheetah.ts [hunk 2]
```diff
@@ -89,15 +89,17 @@ export class Cheetah {
 
   private static _cheetahMutex = new Mutex();
 
+  private _isDetached: boolean = false;
```
Comment (laves) [inline binding/web/src/cheetah.ts:92]: just so we're clear what this means, it should be called `_isWasmMemoryDetached` or something

**Review by laves — CHANGES_REQUESTED** [review]

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/web/src/cheetah.ts [hunk 3]
```diff
@@ -244,10 +246,14 @@ export class Cheetah {
    * @param pcm A frame of audio with properties described above.
    */
   public async process(pcm: Int16Array): Promise<void> {
+    if (this._isDetached) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (ksyeo1010) [reply binding/web/src/cheetah.ts]: the reason this flag is here is to make the error stop showing, as calling process will pop logs indefinitely. The first log they should see is from the `this._errorHandler` function.

[the remark this answers is not in the document]
Comment (ksyeo1010) [reply binding/web/src/cheetah.ts]: The rejecting error isn't also that helpful, 90% of the time this is going to be used in a worker instance, rejecting doesn't help too much. Also the process/flush functions we have are made to return errors via process error callback, all our web sdks do this.

**Review by laves — APPROVED** [review]

**Pull request merged**